prow/github/webhooks.go

```diff
@@ -31,11 +31,6 @@ import (
 func ValidateWebhook(w http.ResponseWriter, r *http.Request, hmacSecret []byte) (string, string, []byte, bool, int) {
 	defer r.Body.Close()
 
-	// Our health check uses GET, so just kick back a 200.
```

Comment:
I am not aware of a deployment that depends on this as a health check endpoint. Regardless, should we wrap this in a flag for backwards compatibility?

Reply:
This is the default livelyness check set up in `cluster/{hook_deployment,start}.yaml`, so if this ceases to exist, Kube will start killing the hook pods.

Whats the rationale for changing this? I do agree that the current state of affairs is not particularly great but then again requiring ppl to do a change their Deployment so hook pods don't get killed is probably worse or WDYT?

Reply:
It appeared to me that the liveness endpoint was `/healthz`, and so I figured this was safe change:
https://github.com/kubernetes/test-infra/blob/6dd4ce833323c861d2f74dfd84cb80abd00aec94/prow/cluster/hook_deployment.yaml#L76

This is not a critical update whatsoever, rather it is an effort to tidy up the `/hook` endpoint so that it doesn't respond to `GET` requests and conforms closer to the Github webhook spec.

Reply:
Oh sorry you are right, forget what I said, seems its too early and I just matched on the starting `h` of the endpoint names :grimacing: 

Looks good then, I think its rather unlikely someone manually edited their deployment to use `/hook`. Maybe still add an announcement thought, just in case?

Reply:
Np. Yes, I agree an announcement is appropriate. I will make the update. Thanks.